modules/post/windows/gather/enum_hyperv_vms.rb

```diff
@@ -30,11 +30,10 @@ def run
     unless have_powershell?
       fail_with(Failure::NoAccess, "The target does not have PowerShell installed so we can't access the state of the Hyper-V VMs")
     end
-    results = psh_exec('Get-VM')
-    if results =~ /is not recognized as the name of a cmdlet/
-      print_error('The target is not a Hyper-V host')
-    elsif results =~ /do not have the required permission/
-      print_error('You need to be running as an elevated admin or a user of the Hyper-V Administrators group to run this module')
+    get_vm = 'try { Get-VM } catch { echo "false" }'
+    results = psh_exec(get_vm)
+    if results == 'false'
+      print_error('Error running `Get-VM` command. Either the target is not a hyper-V host or you do not have enough permissions!')
```

Comment:
Can you catch an exception that's more specific to still identify if the target is not Hyper-V vs a permissions issue? If that information could still be reported to the user, that could really help them. It seems plausible that there could be a permissions-related exception that could be caught at least.

Reply:
yes, actually I tried that but I was not able to catch the permission denied error with any exception, Also I think that the method `psh_exec(cmd)` behaves little different compared to using `cmd_exec('powershell -c #{cmd}')`. When I tried this code to identify the errors differently
```
try { if (Get-VM) { Get-VM} else { echo  "Permission Denied"} catch { echo "Not Present"}
```
it worked fine with `cmd_exec` but not with `psh_exec`.

Reply:
Could we include the original error message from the system in the `print_error`?  It may be regional, but it would be useful.

Reply:
that sounds very useful, I think we could just catch the error and instead of printing not present or permission denied separately we can print the error itself, if that possible.

Reply:
You might be able to do that using another token here, something like:

```ruby
error_token = Rex::Text.rand_text_alphanumeric(8)
get_vm = 'try { Get-VM } catch { echo "#{error_token}"; echo $powershellErrorMessage }'
results = psh_exec(get_vm)
if results.include?(error_token)
  # whoops an error occurred, strip the token and print the message which was the original exception from Powershell
```
This is totally untested pseudo code. I'm not entirely sure how you'd print the exception in PowerShell, but you get the gist.